http/api/transport.go

```diff
@@ -76,14 +79,29 @@ func decodeRequest(_ context.Context, r *http.Request) (interface{}, error) {
 	return msg, nil
 }
 
-// TODO: contact an auth provider
 func authorize(r *http.Request) (string, error) {
 	var apiKey string
+	var id string
+
 	if apiKey = r.Header.Get("Authorization"); apiKey == "" {
 		return "", errUnauthorizedAccess
 	}
 
-	return apiKey, nil
+	authURL := os.Getenv("HTTP_ADAPTER_MANAGER_URL")
```

Comment:
It's better to perform this load & fallback at main, and just pass the configured value to this module.

Reply:
Sure, but how to pass to this module :)

Reply:
Store the client reference to the service itself. Modify [this signature](https://github.com/mainflux/mainflux/blob/master/cmd/http/main.go#L47) and use it as an entrypoint for injecting the client. Then, store the client reference when initializing the transport layer through `MakeHandler` method and use it to authorize requests.